testing/README.md

```diff
@@ -0,0 +1,355 @@
+
+- Use sinon’s spies/ stubs instead of rolling your own version.
+
+  > Why? Spies and stubs provide a consistent interface in case the tests need to be changed, reduce boilerplate so you can focus on the subject under test, can easily be restored to their original versions, and can be tested with more meaningful assertions.
+
+  ```js
+  suite('MyComponent', () => {
+    // bad
+    suite('#myBadFunc', () => {
+      let original;
+      let calledWith;
+
+      setup(() => {
+        spy = function(firstArg) { calledWith = firtArg; }
+        let original = subject.myBadFunc;
+        subject.myBadFunc = spy;
+      });
+
+      teardown(() => {
+        subject.myBadFunc = original;
+      });
+
+      test('it works', () => {
+        subject.doSomething();
+        assert.equal(calledWith, 'some arg');
+      });
+    });
+
+    // good
+    suite('#myGoodFunc', () => {
+      let spy;
+
+      setup(() => {
+        spy = sinon.spy(subject, 'myGoodFunc');
+      });
+
+      teardown(() => {
+        spy.restore();
+      });
+
+      test('it works', () => {
+        subject.doSomething();
+        assert.calledWith(spy, 'some arg');
+      });
+    });
+  });
+  ```
+
+- Use the most meaningful assertion for what you are trying to test. For a list of available assertions, make sure to check out [chai’s `assert` docs](http://chaijs.com/api/assert/).
+
+  > Why? More specific assertions have more specific error messages, making it easier for you (and any developer that deals with this code in the future) to track down and fix test failures.
+
+  ```js
+  // bad
+  assert.isTrue(foo === bar);
+
+  // good
+  assert.equal(foo, bar);
+  ```
+
+- Use sinon’s custom spy assertions, exposed to chai’s `assert` object.
+
+  ```js
+  // somewhere in the test setup...
+  // (see http://sinonjs.org/docs/#assert-expose)
+  sinon.assert.expose(assert);
+
+  // bad
+  assert.isTrue(spy.called);
+
+  // good
+  assert.called(spy);
+  ```
+
+- Always use a test helper that sets up your global test state. This generally includes adding any chai plugins you need, setting up a global sinon sandbox, and any other global hooks you require. It is often useful to expose `sinon`/ `sandbox` and `assert` to the global object in this helper to avoid needing to import them in every test file.
+
+  ```js
+  // Here's an example test-helper file
+
+  import chai, {assert} from 'chai';
+  import sinon from 'sinon';
+  import chaiAsPromised from 'chai-as-promised';
+
+  chai.use(chaiAsPromised);
+  sinon.assert.expose(assert);
+
+  // This creats a sandbox that will automatically restore any
+  // stubbed properties performed during tests.
+  const sandbox = sinon.sandbox.create();
+
+  global.assert = assert;
+  global.sinon = sandbox;
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+  ```
+
+- Use mocha’s ability to automatically finish tests when a returned promise is resolved for tests that are asynchronous and depend on promises.
```

Comment:
Not that I know of, I believe uncaught rejections will be treated as failures with the error message printed out.